src/qs_o3c_methods.F

```diff
@@ -303,6 +319,234 @@ SUBROUTINE calculate_o3c_integrals(o3c, calculate_forces, matrix_p)
+      CALL cp_libint_set_contrdepth(lib, 1)
+
+      DO WHILE (o3c_iterate(o3c_iterator, mepos=mepos) == 0)
+
+         CALL get_o3c_iterator_info(o3c_iterator, mepos=mepos, ikind=ikind, jkind=jkind, &
+                                    kkind=kkind, rij=rij, rik=rik, integral=iabc, tvec=tvec)
+
+         !basis
+         basis_set_a => basis_set_list_a(ikind)%gto_basis_set
+         basis_set_b => basis_set_list_b(jkind)%gto_basis_set
+         basis_set_c => basis_set_list_c(kkind)%gto_basis_set
+         ! center A
+         first_sgfa => basis_set_a%first_sgf
+         la_max => basis_set_a%lmax
+         la_min => basis_set_a%lmin
+         npgfa => basis_set_a%npgf
+         nseta = basis_set_a%nset
+         nsgfa => basis_set_a%nsgf_set
+         sphi_a => basis_set_a%sphi
+         zeta => basis_set_a%zet
+         ! center B
+         first_sgfb => basis_set_b%first_sgf
+         lb_max => basis_set_b%lmax
+         lb_min => basis_set_b%lmin
+         npgfb => basis_set_b%npgf
+         nsetb = basis_set_b%nset
+         nsgfb => basis_set_b%nsgf_set
+         sphi_b => basis_set_b%sphi
+         zetb => basis_set_b%zet
+         ! center C
+         first_sgfc => basis_set_c%first_sgf
+         lc_max => basis_set_c%lmax
+         lc_min => basis_set_c%lmin
+         npgfc => basis_set_c%npgf
+         nsetc = basis_set_c%nset
+         nsgfc => basis_set_c%nsgf_set
+         sphi_c => basis_set_c%sphi
+         zetc => basis_set_c%zet
+
+         ni = SUM(nsgfa)
+         nj = SUM(nsgfb)
+         nk = SUM(nsgfc)
+
+         ALLOCATE (iabc(ni, nj, nk))
+         iabc(:, :, :) = 0.0_dp
+
+         ALLOCATE (tvec(nk, nspin))
+         tvec(:, :) = 0.0_dp
+
+         !need positions for libint. Only relative positions are needed => set ri to 0.0
+         ri = 0.0_dp
+         rj = rij ! ri + rij
+         rk = rik ! ri + rik
+
+         DO iset = 1, nseta
+            ncoa = npgfa(iset)*ncoset(la_max(iset))
+            sgfa = first_sgfa(1, iset)
+            egfa = sgfa+nsgfa(iset)-1
+
+            DO jset = 1, nsetb
+               ncob = npgfb(jset)*ncoset(lb_max(jset))
+               sgfb = first_sgfb(1, jset)
+               egfb = sgfb+nsgfb(jset)-1
+
+               DO kset = 1, nsetc
+                  ncoc = npgfc(kset)*ncoset(lc_max(kset))
+                  sgfc = first_sgfc(1, kset)
+                  egfc = sgfc+nsgfc(kset)-1
+
+                  ALLOCATE (sabc(ncoa, ncob, ncoc))
+                  sabc = 0.0_dp
+
+                  CALL eri_3center(sabc, la_min(iset), la_max(iset), npgfa(iset), zeta(:, iset), &
+                                   ri, lb_min(jset), lb_max(jset), npgfb(jset), zetb(:, jset), &
+                                   rj, lc_min(kset), lc_max(kset), npgfc(kset), zetc(:, kset), &
+                                   rk, lib, op, omega=my_omega, r_cutoff=my_r_cutoff)
+
+                  IF (do_screen) THEN
+                     IF (MAXVAL(ABS(sabc)) < my_eps_screen) THEN
```

Comment:
Screening should be done on the magnitude of the final values of the contracted integrals. The same kind of screening is done in Hartree-Fock but weighted with the contraction coefficients, have a look at the following lines:
https://github.com/cp2k/cp2k/blob/master/src/hfx_energy_potential.F#L544
https://github.com/cp2k/cp2k/blob/master/src/hfx_libint_interface.F#L1613-L1616

Reply:
Yes, indeed. I adapted